src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -1208,11 +1231,11 @@ private void putProperty(
         {
             AmqpStringFW key = amqpStringRW.wrap(valueBuffer, valueOffset.value, valueBuffer.capacity())
                 .set(item.key()).build();
-            valueOffset.value += key.limit();
+            valueOffset.value += key.sizeof();
             AmqpStringFW value = amqpValueRW.wrap(valueBuffer, valueOffset.value, valueBuffer.capacity())
                 .set(item.value()).build();
-            valueOffset.value += value.limit();
-            amqpMapRW.entry(k -> k.setAsAmqpString(key), v -> v.setAsAmqpString(value));
+            valueOffset.value += value.sizeof();
+            applicationPropertyRW.entry(k -> k.setAsAmqpString(key), v -> v.setAsAmqpString(value));
```

Comment:
This style of coding is relying on side effects to update the `applicationPropertyRW` global variable across method invocations and then use the collected result afterwards.

While this is a good technique to take advantage of methods like `applicationProperties.forEach` by passing a consumer lambda, the scope is a bit broad as the variables are global.

Suggest creating inner classes for these cases, such as `AmqpApplicationProperties`, that manages the iteration, and has a local builder RW instance that is initially wrapped, then used for side effects before returning the collected result, all on stack for the same method invocation on `AmqpApplicationProperties`.

Suggest following that same pattern for message properties and annotations as well.